nbviewer/utils.py

```diff
@@ -53,40 +53,26 @@ def url_path_join(*pieces):
         result = '/'
     return result
 
-#def url_rewrite(value):
-#
-#    for reg,template in regs_dict:
-#        matches = reg.match(value)
-#        if matches: 
-#            return template.format(matches.groups)
-
-from collections import OrderedDict
 
-uri_rewrite_dict = OrderedDict()
+def transform_ipynb_uri(uri, uri_rewrite_list):
+    """Transform a given uri (an ipynb 'URI') into an app URL
 
-def transform_ipynb_uri(value, rewrite_providers=None):
-    """Transform a given value (an ipynb 'URI') into an app URL"""
+    State-free part of transforming URIs to nbviewer URLs.
 
-    if not uri_rewrite_dict:
-        from .providers import (
-            default_rewrites,
-            provider_uri_rewrites,
-        )
-        rewrite_providers = rewrite_providers or default_rewrites
-        uri_rewrite_dict.update(provider_uri_rewrites(rewrite_providers))
-
-    for reg, rewrite in uri_rewrite_dict.items():
-        matches = re.match(reg, value)
+    :param uri: uri to transform
+    :param uri_rewrite_list: list of (URI regexes, URL templates) tuples
+    """
+    for reg, rewrite in uri_rewrite_list:
+        matches = re.match(reg, uri)
         if matches:
-            value = rewrite.format(*matches.groups())
-            break
+            return rewrite.format(*matches.groups())
 
     # encode query parameters as last url part
-    if '?' in value:
-        value, query = value.split('?', 1)
-        value = '%s/%s' % (value, quote('?' + query))
+    if '?' in uri:
```

Comment:
Perhaps using urlparse here? considered doing this when i was working on the dropbox links. Not that this is wrong, and clearer.